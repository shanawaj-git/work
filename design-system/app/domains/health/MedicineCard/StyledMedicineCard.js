import styled from 'styled-components';
import { colorSelector } from 'themes';

export const MedicineCardWrapper = styled.div`
  border: 1px solid ${colorSelector('c300')};
  padding: 16px;
  border-radius: 10px;

  .medicine-card-header {
    display: flex;

    img {
      width: 56px;
      height: 56px;
    }

    .card-header-title {
      color: #12121d;
      font-weight: 600;
    }

    .card-header-subTitle {
      color: ${colorSelector('fontPrimary')};
    }
  }

  .medicine-card-details-wrapper {
    display: flex;
    flex-direction: column;
    margin-left: 15px;

    .card-header-subTitle {
      margin-top: 10px;
    }

    .card-header-description {
      color: ${colorSelector('p300')};
    }
  }

  .medicine-card-body {
    background: ${colorSelector('p200')};
    padding: 16px;
    margin-top: 16px;
    border-radius: 16px;
    color: #4c516a;
    line-height: 3;
  }
`;
