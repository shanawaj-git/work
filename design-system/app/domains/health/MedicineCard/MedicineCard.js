import React from 'react';
import PropTypes from 'prop-types';
import Typography, { TypographyType } from 'atoms/Typography';
import { MedicineCardWrapper } from './StyledMedicineCard';

const MedicineCard = props => {
  const { className, image, title, subTitle, description, bodyText } = props;
  return (
    <div className={className}>
      <MedicineCardWrapper>
        <div className="medicine-card-header">
          <img src={image} alt={title} />

          <div className="medicine-card-details-wrapper">
            <div className="medicine-card-header-details">
              <Typography
                typographyType={TypographyType.SUB_HEAD}
                className="card-header-title"
              >
                {title}
              </Typography>

              <Typography
                typographyType={TypographyType.SUB_HEAD}
                className="card-header-subTitle"
              >
                {subTitle}
              </Typography>

              <Typography
                typographyType={TypographyType.SUB_HEAD}
                className="card-header-description"
              >
                {description}
              </Typography>
            </div>
          </div>
        </div>
        <div className="medicine-card-body">
          <p>{bodyText}</p>
        </div>
      </MedicineCardWrapper>
    </div>
  );
};

export default MedicineCard;

MedicineCard.propTypes = {
  className: PropTypes.string,
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  bodyText: PropTypes.string.isRequired,
};
