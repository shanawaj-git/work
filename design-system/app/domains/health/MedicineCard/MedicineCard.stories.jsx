import React from 'react';
import StoryWrapper from 'molecules/utility/StoryWrapper';
import MedicineCard from './MedicineCard';

export default {
  title: 'Components/Containers/cards/MedicineCardComponent',
  component: MedicineCard,
  argTypes: {
    themeType: {
      control: { type: 'select', options: ['DEFAULT', 'MPC'] },
    },
    className: { table: { disable: true } },
  },
};

const cartImage = require('images/pill-icon.png');

const Template = args => (
  <StoryWrapper themeVariant={args.themeType}>
    <MedicineCard {...args} />
  </StoryWrapper>
);

export const PrimaryMedicineCard = Template.bind({});

PrimaryMedicineCard.args = {
  image: cartImage,
  title: 'Hydroxyzine ',
  subTitle: 'Dosage:',
  description: '3 times a day, for 5 days ',
  bodyText: 'Doctors Notes: <br>Take 3 times a day, after food.',
};
