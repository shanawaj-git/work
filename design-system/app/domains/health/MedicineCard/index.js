import MedicineCard from './MedicineCard';

export default MedicineCard;
