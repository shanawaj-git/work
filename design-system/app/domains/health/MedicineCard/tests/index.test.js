import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import MedicineCard from '../index';

const shallowRenderer = new ShallowRenderer();

const pillIcon = require('images/pill-icon.png');

const defaultProps = {
  image: { pillIcon },
  title: 'Hydroxyzine',
  subTitle: 'Dosage:',
  description: '3 times a day, for 5 days',
  bodyText: 'Doctors Notes: <br> Take 3 times a day, after food.',
};

describe('MedicineCard.js', () => {
  describe('index', () => {
    it('Should import succesfully', () => {
      const tree = renderer.create(<MedicineCard {...defaultProps} />);
      expect(tree).toBeTruthy();
    });

    it('Should display Medicine Card in a container with pill image, title, sub title, description and bodyText ', () => {
      shallowRenderer.render(<MedicineCard {...defaultProps} />);
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });
  });
});
