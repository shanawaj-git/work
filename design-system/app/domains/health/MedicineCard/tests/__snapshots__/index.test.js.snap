// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicineCard.js index Should display Medicine Card in a container with pill image, title, sub title, description and bodyText  1`] = `
<div>
  <ForwardRef(StyledMedicineCard__MedicineCardWrapper)>
    <div
      className="medicine-card-header"
    >
      <img
        alt="Hydroxyzine"
        src={
          Object {
            "pillIcon": "IMAGE_MOCK",
          }
        }
      />
      <div
        className="medicine-card-details-wrapper"
      >
        <div
          className="medicine-card-header-details"
        >
          <Typography
            className="card-header-title"
            typographyType="subHead"
          >
            Hydroxyzine
          </Typography>
          <Typography
            className="card-header-subTitle"
            typographyType="subHead"
          >
            Dosage:
          </Typography>
          <Typography
            className="card-header-description"
            typographyType="subHead"
          >
            3 times a day, for 5 days
          </Typography>
        </div>
      </div>
    </div>
    <div
      className="medicine-card-body"
    >
      <p>
        Doctors Notes: &lt;br&gt; Take 3 times a day, after food.
      </p>
    </div>
  </ForwardRef(StyledMedicineCard__MedicineCardWrapper)>
</div>
`;
